### PR TITLE
Add Remoteness to search config and to search url formation

### PR DESCRIPTION
**Pull request by PaulMcInnis**
## Description

This adds `Remoteness` as a search criteria for Indeed and as a scrapable field (for Indeed only as well currently)
#96 

## Context of change

Please add options that are relevant and mark any boxes that apply.

- [x] Software (software that runs on the PC)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Locally on my machine, I'd like to add this to Travis CI but it's a bit iffy as we may not always be guaranteed to get results anywhere.

## Checklist:
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: demo/settings_USA.yaml [hunk 1]
```diff
@@ -42,6 +42,9 @@ search:
   company_block_list:
     - "Infox Consulting"
 
+  # The desired level of work-remoteness (i.e. IN_PERSON, FULLY_REMOTE, ANY)
```
Comment (thebigG) [inline demo/settings_USA.yaml:45]: We should probably document all of the options for remoteness here or somewhere else for users who might not look at the code.

**Review by thebigG — CHANGES_REQUESTED** [review]
Looking good! The only thing I'm concerned about is that it might not be 100% clear to users who don't look at the code all of the remoteness options they have available.

File: demo/settings_USA.yaml (hunk 1 above)
Comment (PaulMcInnis) [reply demo/settings_USA.yaml]: :+1: I've added all the options to this docstring, they are also available by CLI `-h`

**Comment by PaulMcInnis** [comment]
OK - i've expanded the docstring to include all options.

**Review by thebigG — APPROVED** [review]
Awesome :+1:

**Pull request merged**